src/main/scala/beam/router/skim/core/FreightSkimmer.scala

```
@@ -0,0 +1,100 @@
+package beam.router.skim.core
+
+import beam.agentsim.infrastructure.taz.TAZ
+import beam.router.skim.Skims
+import beam.router.skim.readonly.FreightSkims
+import beam.sim.config.BeamConfig
+import beam.utils.matsim_conversion.MatsimPlanConversion.IdOps
+import com.google.inject.Inject
+import com.typesafe.scalalogging.LazyLogging
+import org.matsim.api.core.v01.Id
+import org.matsim.core.controler.MatsimServices
+
+/**
+  * @author Dmitry Openkov
+  */
+class FreightSkimmer @Inject() (
+  matsimServices: MatsimServices,
+  beamConfig: BeamConfig
+) extends AbstractSkimmer(beamConfig, matsimServices.getControlerIO) {
+  import FreightSkimmer._
+  override protected[skim] val readOnlySkim = new FreightSkims
+  override protected val skimFileBaseName: String = FreightSkimmer.fileBaseName
+
+  override protected val skimFileHeader =
+    "tazId,hour,numberOfLoadings,numberOfUnloadings,costPerMile,walkAccessDistanceInM,parkingCostPerHour,observations,iterations"
+  override protected val skimName: String = FreightSkimmer.name
+  override protected val skimType: Skims.SkimType.Value = Skims.SkimType.TC_SKIMMER
+
+  override protected def fromCsv(
```

Should we handle missing data? Or let it crash

---

Same for other skimmers

---

You are right about the possible crash here. Though I think that we need to handle those missed data that may not be written by our skimmer to the file. This `fromCsv` usually read just written files as I understand. It shouldn't contain missed data.